### DISCUSSION TITLE
Distance measurement support

### DISCUSSION THREAD

**Discussion by zokl** [RavenSystem/esp-homekit-devices#1677] [New Features]
Hello,

Thank you very much for your excellent work. I have a question. Is it possible to implement distance measurement into your project? I want to use it for the water level in my well. For now, I have a project based on ESP8266 with a sonic sensor, and I would like to use it in the apple home kit. Is there a way to add support of this kind of sensor to your project?

**Comment by pilot1981** [comment]
Please can you share your actual project? We can use leak sensor of HAA!

**Comment by mariusvasss** [comment]
I think he wants some % measurements...

**Reply by zokl** [reply]
Yes, I need volume in m3 or %.

**Reply by pilot1981** [reply]
> I think he wants some % measurements...

I think this isn’t possibile with HAA
